python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -48,6 +49,57 @@ def destroy_engine(self):
         if self.msg_server is not None:
             self.msg_server.close()
 
+
+    ###########################################################################
+    # panel support (displayed as tabs)
+    def _init_tabs(self):
+        """
+        Create desktop tabs from the registered application panels, following
+        their specified relative positioning.
+        """
+        # make the "Apps" tab appear as a fake panel to fit in the sorting
+        # logic
+        apps_tab = {"callback":   self.desktop_window._register_apps_tab,
+                    "properties": {"position": 0}}
+
+        panels = [apps_tab] + self._engine.panels.values()
+
+        def panel_position(panel):
+            """ Returns the relative position of a panel """
+            # place the panels with unspecified positions at the end
+            LAST_POSITION = float("inf")
```

Comment:
I can think of ways to make it shorter, but at the expense of clarity.  This seems just fine to me.